argus/client/sct/client.py

```diff
@@ -0,0 +1,156 @@
+        )
+
+    def create_resource(self, name: str, resource_type: str, public_ip: str, private_ip: str,
+                        region: str, provider: str, shards_amount: int, state=ResourceState.RUNNING) -> requests.Response:
+        return self.post(
+            endpoint=self.Routes.CREATE_RESOURCE,
+            location_params={"id": str(self.run_id)},
+            body={
+                **self.generic_body,
+                "resource": {
+                    "name": name,
+                    "state": state,
+                    "resource_type": resource_type,
+                    "instance_details": {
+                        "provider": provider,
+                        "region": region,
+                        "public_ip": public_ip,
+                        "private_ip": private_ip,
+                        "shards_amount": shards_amount,
+                    }
+                },
+            }
+        )
+
+    def terminate_resource(self, name: str, reason: str) -> requests.Response:
+        return self.post(
+            endpoint=self.Routes.TERMINATE_RESOURCE,
+            location_params={"id": str(self.run_id), "name": name},
+            body={
+                **self.generic_body,
+                "reason": reason,
+            }
+        )
+
+    def update_shards_for_resource(self, name: str, new_shards: int) -> requests.Response:
+        return self.post(
+            endpoint=self.Routes.UPDATE_SHARDS_FOR_RESOURCE,
+            location_params={"id": str(self.run_id), "name": name},
+            body={
+                **self.generic_body,
+                "shards": new_shards,
+            }
+        )
+
+    def submit_nemesis(self, name: str, class_name: str, start_time: int,
+                       target_name: str, target_ip: str, target_shards: int) -> requests.Response:
+        return self.post(
+            endpoint=self.Routes.SUBMIT_NEMESIS,
+            location_params={"id": str(self.run_id)},
+            body={
+                **self.generic_body,
+                "nemesis": {
+                    "name": name,
+                    "class_name": class_name,
+                    "start_time": start_time,
+                    "node_name": target_name,
+                    "node_ip": target_ip,
+                    "node_shards": target_shards,
+                }
+            }
+        )
+
+    def finalize_nemesis(self, name: str, start_time: int, status: str, message: str) -> requests.Response:
+        return self.post(
+            endpoint=self.Routes.FINALIZE_NEMESIS,
+            location_params={"id": str(self.run_id)},
+            body={
+                **self.generic_body,
+                "nemesis": {
+                    "name": name,
+                    "start_time": start_time,
+                    "status": status,
+                    "message": message,
+                }
+            }
+        )
+
+    def submit_events(self, events: list[dict]) -> requests.Response:
```

Comment:
here also dict is not telling the structure, better use some DTO

Reply:
Done